src/compiler/resolutionCache.ts

```diff
@@ -0,0 +1,316 @@
+/// <reference path="types.ts"/>
+/// <reference path="core.ts"/>
+
+namespace ts {
+    export interface ResolutionCache {
+        setModuleResolutionHost(host: ModuleResolutionHost): void;
+
+        startRecordingFilesWithChangedResolutions(): void;
+        finishRecordingFilesWithChangedResolutions(): Path[];
+
+        resolveModuleNames(moduleNames: string[], containingFile: string, logChanges: boolean): ResolvedModuleFull[];
+        resolveTypeReferenceDirectives(typeDirectiveNames: string[], containingFile: string): ResolvedTypeReferenceDirective[];
+
+        invalidateResolutionOfDeletedFile(filePath: Path): void;
+        invalidateResolutionOfChangedFailedLookupLocation(failedLookupLocation: string): void;
+
+        createHasInvalidatedResolution(): HasInvalidatedResolution;
+
+        clear(): void;
+    }
+
+    type NameResolutionWithFailedLookupLocations = { failedLookupLocations: string[], isInvalidated?: boolean };
+    type ResolverWithGlobalCache = (primaryResult: ResolvedModuleWithFailedLookupLocations, moduleName: string, compilerOptions: CompilerOptions, host: ModuleResolutionHost) => ResolvedModuleWithFailedLookupLocations | undefined;
+
+    /*@internal*/
+    export function resolveWithGlobalCache(primaryResult: ResolvedModuleWithFailedLookupLocations, moduleName: string, compilerOptions: CompilerOptions, host: ModuleResolutionHost, globalCache: string | undefined, projectName: string): ResolvedModuleWithFailedLookupLocations | undefined {

```

Comment:
I think that would be more readable as:
```ts
if (this.getTypeAcquisition().enable) {
    return resolveWithGlobalCache(primaryResult, moduleName, compilerOptions, host,
        this.projectService.typingsInstaller.globalTypingsCacheLocation, this.getProjectName());
}
```
This will avoid that part of the code entirely if there's no global cache.